doc/man3/OSSL_PARAM_get_int.pod

```diff
@@ -0,0 +1,186 @@
+OSSL_PARAM_reserve_buffer, OSSL_PARAM_return_buffer
+- OSSL_PARAM helpers
+
+=head1 SYNOPSIS
+
+=for comment generic
+
+ #include <openssl/params.h>
+
+ OSSL_PARAM *OSSL_PARAM_locate(OSSL_PARAM *p, const char *key);
+
+ #define OSSL_PARAM_TYPE(name)
+
+ int OSSL_PARAM_set_TYPE(OSSL_PARAM *p, const char *key, TYPE *val);
+ int OSSL_PARAM_reserve_TYPE(OSSL_PARAM *p, const char *key TYPE *val);
+ int OSSL_PARAM_reserve_bignum(OSSL_PARAM *p, const char *key,
+                               void *buffer, size_t size);
+ int OSSL_PARAM_reserve_buffer(OSSL_PARAM *p, const char *key,
+                               void *buffer, size_t size);
+
+ int OSSL_PARAM_get_TYPE(OSSL_PARAM *p, const char *key, TYPE *val);
+ int OSSL_PARAM_get_buffer(OSSL_PARAM *p, const char *key,
+                           void **val, size_t *size);
+
+ int OSSL_PARAM_return_TYPE(OSSL_PARAM *p, const char *key, TYPE val);
+ int OSSL_PARAM_return_buffer(OSSL_PARAM *p, const char *key,
+                              void *buff, size_t size);
+
+ int OSSL_PARAM_retrieve_bignum(OSSL_PARAM *p, const char *key,
+                                BIGNUM **bn);
+
+
+=head1 DESCRIPTION
+
+This pages describes a set of utility functions and macros that simplify and
+add type safety to L<OSSL_PARAM(3)> arrays.
+In the following text, B<TYPE> stands for one of the following types:
+
+ int
+ unsigned int ("uint")
+ int64_t
+ uint64_t
+ long
+ unsigned long ("ulong")
+ size_t
+ double
+ void * ("pointer")
+ array of bytes ("buffer")
+ BIGNUM
+
+A pointer is a generic pointer. All of the types above are fixed-size,
+except for C<buffer> and C<BIGNUM>.
+
+OSSL_PARAM_locate() finds the parameter in the B<p> list with the name
+B<key>.  Most of the functions described below take a pointer to a parameter
+list, B<p>, and a key naming a specific parameter, B<key>.  If B<key>
+is NULL, then B<p> is used directly, otherwise OSSL_PARAM_locate() is
+called to find a matching entry for B<key>.  This technique can be used
+for optional parameters, or for passing values when the array has been
+initialized with fixed positions:
+
+ OSSL_PARAM plist[] = {
+     OSSL_PARAM_int("optional"),
+     OSSL_PARAM_pointer("ptr"),
+     OSSL_PARAM_bignum("prime"),
+     OSSL_PARAM_int("status"),
+     { NULL }
+ };
+
+ if ((p = OSSL_PARAM_locate(plist, "optional")) != NULL)
+     OSSL_PARAM_get_int(p, NULL, &anint);
+
+ OSSL_PARAM_get_int(&plist[0], NULL, &anint);
+ OSSL_PARAM_get_pointer(&plist[1], NULL, &ptr);
+ ...
+
+The macro OSSL_PARAM_TYPE() (for example, OSSL_PARAM_int()) returns an
+initialized value of an B<OSSL_PARAM> array initialized for the specified
+B<TYPE>.
+
+In the following paragraphs, consider one function calling another, and
+that B<OSSL_PARAM> is used to pass parameters between them:
+
+ /* For simplicity, showing only one argument */
+ called_function(OSSL_PARAM *p)
+ {
+ }
+
+ calling_function()
+ {
+     called_function(plist);
+ }
```

Comment:
This suggested code implies that `OSSL_PARAM` would be used as a generic replacement for normal C function parameters.

I am firmly opposed to such an implication, and that alone will get a -1 from me.

Reply:
The comment above ```called_function``` tries to imply otherwise.  How can I make that more clear?